$OpenBSD: patch-src_video_x11_SDL_x11window_c,v 1.3 2017/07/06 05:32:32 landry Exp $

The Xlib documentation demands that 32-bit values here be passed in a long,
even when long itself isn't a 32-bit value. Otherwise libx11 might read
memory incorrectly (Bugzilla #3692).

97c66d81b2c9 upstream.

Index: src/video/x11/SDL_x11window.c
--- src/video/x11/SDL_x11window.c.orig
+++ src/video/x11/SDL_x11window.c
@@ -376,7 +376,7 @@ X11_CreateWindow(_THIS, SDL_Window * window)
     Atom _NET_WM_WINDOW_TYPE;
     Atom wintype;
     const char *wintype_name = NULL;
-    int compositor = 1;
+    long compositor = 1;
     Atom _NET_WM_PID;
     Atom XdndAware, xdnd_version = 5;
     long fevent = 0;
